src/amoc_config.erl

```diff
@@ -0,0 +1,67 @@
+%%==============================================================================
+%% Copyright 2016 Erlang Solutions Ltd.
+%% Licensed under the Apache License, Version 2.0 (see LICENSE file)
+%%==============================================================================
+-module(amoc_config).
+
+-export([start/0,
+         set/2,
+         get/1,
+         get/2,
+         fetch/1]).
+
+%% For tests
+-export([set_env_variables/2]).
+
+%% ------------------------------------------------------------------
+%% API
+%% ------------------------------------------------------------------
+-spec start() -> ok.
+start() ->
+    ok = set_env_variables(os:getenv()).
+
+-spec set(atom(), any()) -> ok.
+set(Name, Value) ->
+    ok = application:set_env(amoc, Name, Value).
+
+-spec get(atom()) -> any().
+get(Name) ->
+    get(Name, undefined).
+
+-spec get(atom(), any()) -> any().
+get(Name, Default) ->
+    application:get_env(amoc, Name, Default).
+
+-spec fetch(atom()) -> any().
+fetch(Name) ->
+    case application:get_env(amoc, Name) of
+        {ok, Value} -> Value;
+        undefined -> throw({key_not_found, Name})
+    end.
+
+%% ------------------------------------------------------------------
+%% Internal Function Definitions
+%% ------------------------------------------------------------------
+-spec set_env_variables([string()]) -> ok.
+set_env_variables(EnvVars) ->
+    set_env_variables(EnvVars, fun(_) -> true end).
+
+-spec set_env_variables([string()], fun((string()) -> boolean())) -> ok.
+set_env_variables(EnvVars, Filter) ->
+    Vars = parse_variables(EnvVars),
+    [ ok = set(Name, Value) || {Name, Value} <- Vars, Filter(Name) ],
+    ok.
+
+-spec parse_variables([string()]) -> [{atom(), any()}].
+parse_variables(EnvVars) ->
+    Vars = [ re:split(Var, "=", [{return, list}, {parts, 2}])
+             || Var <- EnvVars ],
+    %% list_to_atom should be safe here, as we are in control of env variables
```

Comment:
If we want to use a custom setting (like list of xmpp servers or whatever that makes sense only inside a scenario) then we would need to define it there as well.
The goal of introducing this is to allow Docker container configuration done only via environmental variables.
Let's start with that, that I'm not sure if this is even a good approach :)